### DISCUSSION TITLE
Pulsar Requirements

### DISCUSSION THREAD

**Discussion by zirota** [milvus-io/milvus#26016] [Q&A and General discussion]
Hello may I know what are recommended configurations for the Pulsar supporting Milvus?

- Retention policies - should I use size-based or time-based?
- Partitions - how many partitions should the topic be created with?
- Can I enable topic compaction? What size should I compact on?

Also, are there any size estimates for a typical pulsar cluster supporting Milvus with around 25 collections?

**Comment by LoveEachDay** [comment]
@zirota Here's some recommendations for pulsar:
1. use time-based retention, by default 7 days, you may change to 1day
2. no partitions (milvus don't support topic partitions)
3. not enable topic compaction

**Reply by zirota** [reply]
Thank you @LoveEachDay.

As topics are created by Milvus, are we able to configure these values from Milvus side? Or must i explicitly set it in the Pulsar server configurations?
